fedimint-core/src/encoding/mod.rs

```diff
@@ -564,10 +499,9 @@ where
     fn consensus_encode<W: std::io::Write>(&self, writer: &mut W) -> Result<usize, std::io::Error> {
         if TypeId::of::<T>() == TypeId::of::<u8>() {
             // unsafe: we've just checked that T is `u8` so the transmute here is a no-op
-            return consensus_encode_bytes_static(
-                unsafe { mem::transmute::<&[T; SIZE], &[u8; SIZE]>(self) },
-                writer,
-            );
+            let bytes = unsafe { mem::transmute::<&[T; SIZE], &[u8; SIZE]>(self) };
```

Comment:
OK. Inlining might be better.

Reply:
FYI this might also be transmuting lifetimes